ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSpec.java

```diff
@@ -437,4 +448,30 @@ public void setNeedDupCopyCheck(boolean isFirstIncPending) {
     // Check HIVE-21197 for more detail.
     this.needDupCopyCheck = isFirstIncPending;
   }
+
+  public boolean isPathOwnedByHive() {
+    return isPathOwnedByHive;
+  }
+
+  public static boolean isPathOwnedByHive(HiveConf conf, String user) {
+    String ownerName = conf.get(HiveConf.ConfVars.STRICT_MANAGED_TABLES_MIGRARTION_OWNER.varname, "hive");
+    return  (user == null || ownerName.equals(user));
+  }
+
+  public void setPathOwnedByHive(boolean pathOwnedByHive) {
+    isPathOwnedByHive = pathOwnedByHive;
+  }
+
+  public void setPathOwnedByHive(HiveConf conf, String user) {
```

Comment:
This method also can be moved to ReplUtils and use setPathOwnedByHive(boolean) method to set it in ReplicationSpec. 
Just to decouple the logic outside.

Reply:
as the member is part of replication scope ..its better to keep it in replication scope only